erpnext/stock/get_item_details.py

```diff
@@ -257,25 +259,52 @@ def insert_item_price(args):
 				"price_list": args.price_list,
 				"item_code": args.item_code,
 				"currency": args.currency,
-				"price_list_rate": price_list_rate
+				"price_list_rate": price_list_rate,
+				"min_qty": args.qty
 			})
 
-			name = frappe.db.get_value('Item Price', {'item_code': args.item_code, 'price_list': args.price_list, 'currency': args.currency}, 'name')
+			name = frappe.db.get_value(
+					'Item Price',
+					{
+						'item_code': args.item_code,
+						'price_list': args.price_list,
+						'currency': args.currency,
+						'min_qty': args.qty
+					},
+					'name')
 
 			if name:
 				item_price = frappe.get_doc('Item Price', name)
 				item_price.price_list_rate = price_list_rate
 				item_price.save()
-				frappe.msgprint(_("Item Price updated for {0} in Price List {1}").format(args.item_code,
-					args.price_list))
+				frappe.msgprint(_("Item Price updated for {0} in Price List {1} with Mininum Quantity {2}").format(args.item_code,
+					args.price_list, args.qty))
 			else:
 				item_price.insert()
-				frappe.msgprint(_("Item Price added for {0} in Price List {1}").format(args.item_code,
-					args.price_list))
+				frappe.msgprint(_("Item Price added for {0} in Price List {1}w ith Mininum Quantity {2}").format(args.item_code,
+					args.price_list, args.qty))
 
-def get_price_list_rate_for(price_list, item_code):
-	return frappe.db.get_value("Item Price",
-			{"price_list": price_list, "item_code": item_code}, "price_list_rate")
+def get_price_list_rate_for(price_list, item_code, qty):
+	"""
+		Return Price Rate based on min_qty of each Item Price Rate.\
+		For example, desired qty is 10 and Item Price Rates exists
+		for min_qty 9 and min_qty 20. It returns Item Price Rate for qty 9 as
+		the best fit in the range of avaliable min_qtyies
+
+		:param price_list: str (Standard Buying or Standard Selling)
+		:param item_code: str, Item Doctype field item_code
+		:param qty: Derised Qty
+	"""
+	price_list_rate = frappe.db.sql("""
+		select price_list_rate
+		from `tabItem Price`
+		where item_code=%s
+			and price_list=%s
+			and min_qty<=%s order by min_qty desc""",
```

Comment:
Test cases missing for this - make a purchase order and see different prices are selected for different qtys

Also show UI screenshot to show it works from the UI